docs/framework/30scoring/implementation.md

```diff
@@ -1,51 +1,187 @@
-[Secret Storage](../20categories/30implementation/secret-storage.md),
-[Product Development Maturity](../20categories/30implementation/maturity.md),
-[Well Known Platform](../20categories/30implementation/known-platform.md),
-[Well Known Smart Contract Programming Language](../20categories/30implementation/known-language.md),
-[Formal Verification](../20categories/30implementation/formal-verification.md),
-[Testing](../20categories/30implementation/testing.md),
-[Open Source Code](../20categories/30implementation/open-source.md),
-[Verified Code on Block Explorer](../20categories/30implementation/verified-code.md),
-[Documentation](../20categories/30implementation/documentation.md),
-[Bug Bounty](../20categories/30implementation/bug-bounty.md),
-[Ability to Pause Project](../20categories/30implementation/pause.md),
-and 
-[Ability to Ban Addresses](../20categories/30implementation/ban-address.md).
+The equation for the *Implementation Risk Score* is:
 
+```
+Implementation Risk Score = Maximum of (Reducing Implementation Risk Score, Uncovering Extant Risk Score, and Responding to Materialized Risk Score)
+```
 
-The equation for the *Implementation Risk Score* is:
+### Reducing Implementation Risk
+```
+Reducing Implementation Risk Score = 
+ Mixing of Control and Data Plane Risk Score +
+ Role Based Access Control Risk Score +
+ Upgradable Risk Score +
+ Secret Storage Risk Score +
+ Product Development Maturity Risk Score +
+ Well Known Platform Risk Score +
+ Well Known Smart Contract Programming Language Risk Score
+```
+
+#### Mixing of Control and Data Plane Risk Score
+Mixing of Control and Data Plane is described in the 
+[Mixing of Control and Data Plane](../20categories/30implementation/mixing-control-data-flow.md) section. If the Control and Data Plane are mixed, then the Mixing of Control and Data Plane Risk Score is 20, or 0 otherwise.
+
+The Control and Data Plane are considered mixed if the answer to any of the following questions is yes:
+
+* Can Control Plane functions be called from Data Plane functions without some additional authentication checks?
+* If Control Plane messages are sent across the crosschain communications channel, are Control Plane messages signed by the same keys as Data Plan messages?
+
+
+#### Role Based Access Control Risk Score
+Role Based Access Control is described in the [Role Based Access Control](../20categories/30implementation/access-control.md) section.
+
+The Role Based Access Control Risk Score depends on the permissioned control / ownership of the bridge contracts. If there are multiple contracts, and they  are configured differently, then score the highest score.
+
+* There is no permissioned ownership feature: Score 0.
+* There a single EOA as the owner: Score 10.
+* Role Based Access Control has been implemented: Score 0.
+* There is no Role Based Access Control, but a threshold number of signers need to approve administrative actions: Score 5.
```

Comment:
should the size of the signer set and the threshold be a differentiating factor? is a 2 of 3 as risky as a 14 of 20?

Reply:
I have factored this in, in an updated PR